test/einops_comparison.jl

```diff
@@ -0,0 +1,182 @@
+""" These tests are inspired by the paper published on python's "einops" package.
+
+See 'Alex Rogozhnikov, 2022. "EINOPS: CLEAR AND RELIABLE TENSOR MANIPULATIONS WITH EINSTEIN-LIKE NOTATION",
+published at ICLR 2022'; Listing 1.
+https://openreview.net/pdf?id=oapKSVM2bcj
+"""
+
+@testset "Einops paper - Listing 1" begin
+
+  # Format:
+  # NUMBER NUMPY-CODE
+  #        EINOPS-CODE
+  # JULIA-CODE...
+
+  # 1 np.transpose(x, [0, 3, 1, 2]) 
+  #   rearrange(x, 'b h w c -> b c h w')
+  @testset "Test 1" begin
+    x = rand(3, 4, 5, 6)
+    @test (
+      permutedims(x, [1, 4, 2, 3]) 
+      == @cast _[b, c, h, w] := x[b, h, w, c])
+  end
+
+  # 2 np.reshape(x, [x.shape[0]*x.shape[1], x.shape[2]]) 
+  #   rearrange(x, 'h w c -> (h w) c ') 
+  @testset "Test 2" begin
+    x = rand(3, 4, 5)
+    @test (
+      reshape(x, size(x, 1) * size(x, 2), size(x, 3)) == 
+      @cast _[(h, w), c] := x[h, w, c]
+    )
+  end
+
+  # 3 np.squeeze(x, 0) 
+  #   rearrange(x, '() h w c -> h w c') 
+  @testset "Test 3" begin
+    x = rand(1, 2, 3, 4)
+    @test (
+      reshape(x, 2, 3, 4) ==
+      @cast _[h, w, c] := x[_, h, w, c])
+  end
+
+  # 4 np.expand_dims(x, -1) 
+  #   rearrange(x, 'h w c -> h w c ()') 
+  @testset "Test 4" begin
+    x = rand(2, 3, 4)
+    @test (
+      reshape(x, 2, 3, 4, 1) ==
+      @cast _[h, w, c, 1] := x[h, w, c])
+  end
+
+  # 5 np.stack([r, g, b], axis=2) 
+  #   rearrange([r, g, b], 'c h w -> h w c') 
+  @testset "Test 5" begin
+    xs = [rand(2, 2) for _ in 1:3]
+    @test (
+      cat(xs..., dims=3) ==
+      @cast _[h, w, i] := xs[i][h, w])
+  end
+
+  # 6 np.concatenate([r, g, b], axis=0) 
+  #   rearrange([r, g, b], 'c h w -> (c h) w')
+  @testset "Test 6" begin
+    xs = [rand(2, 2) for _ in 1:3]
+    expand_first(x) = reshape(x, 1, size(x)...)
+    @test (
+      cat(expand_first.(xs)..., dims=1) ==
+      @cast _[i, h, w] := xs[i][h, w]
+    )
+  end
+
+  # 7 np.flatten(x) 
+  #   rearrange(x, 'b t c -> (b t c) ') 
+  @testset "Test 7" begin
+    x = rand(3, 4, 5)
+    @test (
+      x[:] == 
+      @cast _[(b, t, c)] := x[b, t, c]
```

Comment:
```suggestion
      x[:] == vec(x) == 
      @cast _[(b, t, c)] := x[b, t, c]
```

Reply:
Done in 2946fca2618522d4b84cc5a8231b8cab69ed92c5.